src/axolotl/datasets.py

```diff
@@ -148,13 +156,17 @@ def __iter__(self):
                             input_ids, dtype=self.tokens_dtype
                         )
                         attention_mask_with_concat = torch.tensor(
-                            attention_mask, dtype=self.tokens_dtype
+                            [idx * m for m in attention_mask], dtype=torch.int16
```

Comment:
this is needed to create the correct expanded attention mask when using the legacy packing and SDP or flash-attn